opentelemetry/src/global/propagation.rs

```diff
@@ -1,19 +1,29 @@
 use crate::propagation::TextMapPropagator;
 use crate::trace::noop::NoopTextMapPropagator;
-use once_cell::sync::Lazy;
-use std::sync::RwLock;
+use std::sync::{OnceLock, RwLock};
 
 /// The current global `TextMapPropagator` propagator.
-static GLOBAL_TEXT_MAP_PROPAGATOR: Lazy<RwLock<Box<dyn TextMapPropagator + Send + Sync>>> =
-    Lazy::new(|| RwLock::new(Box::new(NoopTextMapPropagator::new())));
+static GLOBAL_TEXT_MAP_PROPAGATOR: OnceLock<RwLock<Box<dyn TextMapPropagator + Send + Sync>>> =
+    OnceLock::new();
 
 /// The global default `TextMapPropagator` propagator.
-static DEFAULT_TEXT_MAP_PROPAGATOR: Lazy<NoopTextMapPropagator> =
-    Lazy::new(NoopTextMapPropagator::new);
+static DEFAULT_TEXT_MAP_PROPAGATOR: OnceLock<NoopTextMapPropagator> = OnceLock::new();
+
+/// Ensures the `GLOBAL_TEXT_MAP_PROPAGATOR` is initialized with a `NoopTextMapPropagator`.
```

Comment:
nit: For each of these methods, we could use a comment similar to what we have for Baggage:

```rust
/// Returns the default baggage, ensuring it is initialized only once.
````

So, something like:
```
/// Returns the global TextMapPropagator ensuring it is initialized only once
```